_rules/image-button-accessible-name-59796f.md

```diff
@@ -55,128 +55,105 @@ There is a known combination of a popular browser and assistive technology that
 
 #### Passed Example 1
 
-Image button element with [accessible name][] through `alt` attribute
+The image button has an [accessible name][] through the `alt` attribute.
 
 ```html
-<input type="image" name="submit" src="button.gif" alt="Submit" />
+<input type="image" src="/test-assets/shared/search-icon.png" alt="Search" />
 ```
 
 #### Passed Example 2
 
-Image button element with [accessible name][] through `aria-label`
+The image button has an [accessible name][] through the `aria-label` attribute.
 
 ```html
-<input type="image" name="submit" src="button.gif" aria-label="Submit" />
+<input type="image" src="/test-assets/shared/search-icon.png" aria-label="Search" />
 ```
 
 #### Passed Example 3
 
-Image button element with [accessible name][] through `title` attribute
+The image button has an [accessible name][] through the `title` attribute.
 
 ```html
-<input type="image" name="submit" src="button.gif" title="Submit" />
+<input type="image" src="/test-assets/shared/search-icon.png" title="Search" />
 ```
 
 #### Passed Example 4
 
-Image button element with [accessible name][] through `aria-labelledby`
+The image button has an [accessible name][] through the `aria-labelledby` attribute.
 
 ```html
-<input type="image" name="submit" src="button.gif" aria-labelledby="id1" />
-<div id="id1">Submit</div>
-```
-
-#### Passed Example 5
-
-[accessible name][] is not empty.
-
-```html
-<input type="image" name="submit" src="button.gif" alt=":-)" />
-```
-
-#### Passed Example 6
-
-Image button element with [accessible name][] through `alt` attribute
-
-```html
-<input type="image" name="submit" src="button.gif" alt="123" />
+<input type="image" src="/test-assets/shared/search-icon.png" aria-labelledby="id1" />
+<div id="id1">Search</div>
 ```
 
 ### Failed
 
 #### Failed Example 1
 
-Image button element with empty [accessible name][].
+The image button element has an empty [accessible name][]. The `name` attribute can not be used to provide an [accessible name][].
 
 ```html
-<input type="image" name="submit" src="button.gif" />
+<input type="image" name="search" src="/test-assets/shared/search-icon.png" />
```

Comment:
While the input can't get its accessible name from the name attribute, fallback for input type="image" is the literal string "Submit Query", so this still has an accessible name and should pass.
See step 4 of:
https://w3c.github.io/html-aam/#input-type-image

Reply:
To make things a bit more complicated, Step 4 actually has:
"…user agent **may**…"
which makes it more squishy. In testing the above html I found:
Safari OSX VoiceOver – “Submit Button”
Chrome OSX VoiceOver – “Submit Button”
Chrome Win10 NVDA – “Button Submit”
Chrome Win10 JAWS – “Submit Button”
Edge Win10 NVDA – “Button Blank”
Edge Win10 JAWS – “Button Blank”
FF Win10 NVDA – “Button Submit Query”
FF Win10 JAWS – “Submit Query Button”
So it would appear only Edge is not taking the "may" approach. This is problematic. Whether an image input lack of an accessible name is somewhat inconsistent and flagging the markup appears to be a false positive in several browser/AT stacks.

Reply:
Thank you for providing this test data. Would you mind opening a separate issue or pull request for this? The purpose of this PR is to improve the test case descriptions. I don't think your comment relates to this.

Reply:
I'm going to merge this. I'll open a separate issue.